dipy/reconst/ivim.py

```diff
@@ -153,20 +166,18 @@ def nlls_fit(data, gtab, guess_params):
 
     Returns
     -------
-    nlls_params: the S0, f, D_star, D value for each voxel.
+    ivim_params: the S0, f, D_star, D value for each voxel.
 
     """
     flat_data = data.reshape((-1, data.shape[-1]))
     # Flatten for the iteration over voxels:
     bvals = gtab.bvals
     ivim_params = np.empty((flat_data.shape[0], 4))
     for vox in range(flat_data.shape[0]):
-        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox],
-                               guess_params[vox])
-        ivim_params[vox, :4] = popt
+        res = leastsq(_ivim_error,
```

Comment:
@arokem If I replace this with minimize(_ivim_error, x0[vox], args=(bvals, flat_data[vox]) ) I get the error : 

ValueError: setting an array element with a sequence.

```python
File "/home/shahnawaz/dev/gsoc16/dipy/dipy/reconst/tests/test_ivim.py", line 33, in test_fit
    ivim_fit = ivim_model.fit(data)
  File "/home/shahnawaz/dev/gsoc16/dipy/dipy/reconst/ivim.py", line 111, in fit
    x0)
  File "/home/shahnawaz/dev/gsoc16/dipy/dipy/reconst/ivim.py", line 179, in one_stage
    args=(bvals, flat_data[vox]))
  File "/usr/local/lib/python2.7/dist-packages/scipy/optimize/_minimize.py", line 441, in minimize
    return _minimize_bfgs(fun, x0, args, jac, callback, **options)
  File "/usr/local/lib/python2.7/dist-packages/scipy/optimize/optimize.py", line 847, in _minimize_bfgs
    gfk = myfprime(x0)
  File "/usr/local/lib/python2.7/dist-packages/scipy/optimize/optimize.py", line 289, in function_wrapper
    return function(*(wrapper_args + args))
  File "/usr/local/lib/python2.7/dist-packages/scipy/optimize/optimize.py", line 622, in approx_fprime
    return _approx_fprime_helper(xk, f, epsilon, args=args)
  File "/usr/local/lib/python2.7/dist-packages/scipy/optimize/optimize.py", line 562, in _approx_fprime_helper
    grad[k] = (f(*((xk + d,) + args)) - f0) / d[k]
ValueError: setting an array element with a sequence.
```